ee/server/calendar/google_calendar.go

```diff
@@ -579,17 +593,26 @@ func (c *GoogleCalendar) createEvent(
 		return nil, ctxerr.Wrap(c.config.Context, err, "creating Google calendar event")
 	}
 
-	// Watch for event changes
-	secondsToEventEnd := eventEnd.Sub(now).Milliseconds() / 1000
-	eventUUID := uuid.New().String()
-	channelUUID := uuid.New().String()
-	resourceID, err := c.config.API.Watch(eventUUID, channelUUID, uint64(secondsToEventEnd))
-	if err != nil {
-		return nil, ctxerr.Wrap(c.config.Context, err, "watching Google calendar event")
+	// Watch for event changes, if not already watching.
+	var eventUUID, channelID, resourceID string
+	if opts.EventUUID == "" || opts.ChannelID == "" || opts.ResourceID == "" {
+		// Watch for changes until the end of the event, plus 1 more week. The extra time is to handle cases when end user moves the event forward.
```

Comment:
what happens when a user moves an event too far?  do we create a new event?  if so, what happens when that event comes into a week timeframe?

Reply:
Nothing happens. We have a story to limit the timeframe: https://github.com/fleetdm/fleet/issues/20177

We watch the event for this long. It doesn't matter where the user moves it. Normally, we expect the event to fire its webhook and we stop watching it. But if it has been a week since the original event time, we fall back to checking it every 30 minutes and we should not see callbacks anymore.